### PR TITLE
Prow jobs explicitly set it's own resources requests/limit

### PR DESCRIPTION
**Pull request by chaodaiG**
This removes the unnecessary resource allocation of:
- janitor pods
- maintenance jobs

Other than these, some of the jobs don't really need to use that much cpu and memory, this enables us tweaking them on a per job basis

/assign chizhg

Related to: #1734 
/cc coryrc

### CODE REVIEW

File: config/build-cluster/cluster/200-cpu-limitrange.yaml [hunk 1]
```diff
@@ -1,26 +0,0 @@
...
-# limitations under the License.
-
-apiVersion: v1
-kind: LimitRange
-metadata:
-  name: cpu-limit-range
-  namespace: test-pods
-spec:
-  limits:
-    - default:
```
Comment (coryrc) [inline config/build-cluster/cluster/200-cpu-limitrange.yaml:22]: I'd rather keep these sensible defaults

File: tools/config-generator/main.go [hunk 2]
```diff
@@ -440,12 +462,12 @@ func newbaseProwJobTemplateData(repo string) baseProwJobTemplateData {
 	data.Args = make([]string, 0)
 	data.Volumes = make([]string, 0)
 	data.VolumeMounts = make([]string, 0)
-	data.Resources = make([]string, 0)
 	data.Env = make([]string, 0)
 	data.ExtraRefs = []string{"- org: " + data.OrgName, "  repo: " + data.RepoName}
 	data.Labels = make([]string, 0)
 	data.Optional = ""
 
+	setReourcesReqForJob(defaultResource, &data)
```
Comment (coryrc) [inline tools/config-generator/main.go:470]: function name is misspelled

File: tools/config-generator/main.go [hunk 3]
```diff
@@ -177,6 +177,28 @@ var (
 	emittedOutput bool
 
 	projNameRegex = regexp.MustCompile(`.+-[0-9\.]+$`)
+
+	// defaultResource setting for each job
```
Comment (coryrc) [inline tools/config-generator/main.go:181]: Lacking CPU requests

File: config/build-cluster/cluster/200-cpu-limitrange.yaml (hunk 1 above)
Comment (chaodaiG) [reply config/build-cluster/cluster/200-cpu-limitrange.yaml:22]: Do you mean cpu or both?

File: tools/config-generator/main.go (hunk 3 above)
Comment (chaodaiG) [reply tools/config-generator/main.go:181]: done

File: tools/config-generator/main.go (hunk 2 above)
Comment (chaodaiG) [reply tools/config-generator/main.go]: fixed

File: tools/config-generator/main.go (hunk 3 above)
Comment (chaodaiG) [reply tools/config-generator/main.go:181]: added

File: config/build-cluster/cluster/200-cpu-limitrange.yaml (hunk 1 above)
Comment (coryrc) [reply config/build-cluster/cluster/200-cpu-limitrange.yaml:22]: Both. It causes very weird intermittent problems when the default requests (.1cpu) are used by some jobs. It's just nice prevention-in-depth to keep.

Comment (chaodaiG) [reply config/build-cluster/cluster/200-cpu-limitrange.yaml:22]: For lightweight jobs I actually would prefer it uses the default .1cpu as request, as well as any other pods such as janitor

Comment (chaodaiG) [reply config/build-cluster/cluster/200-cpu-limitrange.yaml:22]: The other problem is that sidecar in each Prow job pod also requests these resources, which is pretty confusing

Comment (chaodaiG) [reply config/build-cluster/cluster/200-cpu-limitrange.yaml:22]: So when we say each container requests 2 CPUs, a Prow job actually requests 4 CPUs

Comment (coryrc) [reply config/build-cluster/cluster/200-cpu-limitrange.yaml:22]: Why don't we have resource requests on those?
I also think our janitors need a little more than .1 cpu, but we can reduce their quantity now that they've been fixed.

Comment (chaodaiG) [reply config/build-cluster/cluster/200-cpu-limitrange.yaml:22]: Prow job sidecar is really lightweight, it shouldn't need 2 CPUs. If janitors need a little more than .1 cpu we can grant it there

Comment (coryrc) [reply config/build-cluster/cluster/200-cpu-limitrange.yaml:22]: I asked someone to double-check, but looks like the sidecar is a problem. Could you boost the janitor CPU allocation slightly and reduce their number at the same time? The latter could be delayed if you want.

Comment (chaodaiG) [reply config/build-cluster/cluster/200-cpu-limitrange.yaml:22]: done

Comment (coryrc) [reply config/build-cluster/cluster/200-cpu-limitrange.yaml:22]: Ah, we had a misunderstanding. The sidecar shares the resources dedicated to the pod, so it doesn't hurt to keep the defaults.

**Comment by coryrc** [comment]
/lgtm

I guess all the jobs have resources now so that should be good enough.

**Comment by chaodaiG** [comment]
> /lgtm
> 
> I guess all the jobs have resources now so that should be good enough.

Per the offline discussion, we can explicitly set the sidecar requests, will skip setting it on every single job then

/hold

File: config/build-cluster/cluster/200-memory-limitrange.yaml [hunk 4]
```diff
@@ -22,5 +22,5 @@ spec:
     - default:
         memory: 8Gi
       defaultRequest:
-        memory: 4Gi
+        memory: 8Gi
```
Comment (chizhg) [inline config/build-cluster/cluster/200-memory-limitrange.yaml:25]: What's the reason for changing this?

Comment (chaodaiG) [reply config/build-cluster/cluster/200-memory-limitrange.yaml:25]: It used to be sidecar 4Gb, test container 4Gb, so actually a pod requests 8Gb. Now that we set sidecar to be real low, so we can make the real request of 8Gb

Comment (chizhg) [reply config/build-cluster/cluster/200-memory-limitrange.yaml:25]: I see, great improvements on resource utilization!

**Pull request merged**